test/dotnet/src/PrepareExecBinding.cs

```diff
@@ -126,29 +128,47 @@ public static DbCommand SetBindParams(string[] result,
 								string geoWKT = arguments[0];
 								int srid = (int)GetSqlDbValue("int", arguments[1]);
 								
-								((SqlParameter)sqlCmd.Parameters[param[1].Trim()]).SqlDbType = SqlDbType.Udt;
-								((SqlParameter) sqlCmd.Parameters[param[1].Trim()]).UdtTypeName = "Geometry";
+								// Create a WKT reader and parse the geometry
+								var reader = new WKTReader();
+								var geometry = reader.Read(geoWKT);
+								geometry.SRID = srid;
+
+								// Convert to SQL Server format
+								var sqlGeometryWriter = new SqlServerBytesWriter();
+								byte[] sqlServerBytes = sqlGeometryWriter.Write(geometry);
 
-								sqlCmd.Parameters[param[1].Trim()].Value = SqlGeometry.STGeomFromText(new SqlChars(new SqlString(geoWKT)), srid);
-								sqlCmd.Parameters[param[1].Trim()].Size = 65535;
+								// Set up the parameter
+								((SqlParameter)sqlCmd.Parameters[param[1].Trim()]).SqlDbType = SqlDbType.Udt;
+								((SqlParameter)sqlCmd.Parameters[param[1].Trim()]).UdtTypeName = "geometry";
+								sqlCmd.Parameters[param[1].Trim()].Value = sqlServerBytes;
+								sqlCmd.Parameters[param[1].Trim()].Size = -1;
 							}
 								break;
 							case "geography":
 							{
 								if (ConfigSetup.Database.Equals("oledb", StringComparison.InvariantCulture))
 								{
 									testUtils.PrintToLogsOrConsole("GEOGRAPHY NOT SUPPORTED BY OLEDB", logger, "error");
-									break;
+									break;    
 								}
 								string[] arguments = param[2].Split(':', 2);
 								string geoWKT = arguments[0];
 								int srid = (int)GetSqlDbValue("int", arguments[1]);
-														
-								((SqlParameter)sqlCmd.Parameters[param[1].Trim()]).SqlDbType = SqlDbType.Udt;
-								((SqlParameter)sqlCmd.Parameters[param[1].Trim()]).UdtTypeName = "Geography";
 
-								sqlCmd.Parameters[param[1].Trim()].Value = SqlGeography.STGeomFromText(new SqlChars(new SqlString(geoWKT)), srid);
-								sqlCmd.Parameters[param[1].Trim()].Size = 65535;
+								// Create a WKT reader and parse the geometry
+								var reader = new WKTReader();
+								var geometry = reader.Read(geoWKT);
+								geometry.SRID = srid;
+
+								// Convert to SQL Server format
+								var sqlGeometryWriter = new SqlServerBytesWriter();
```

Comment:
SqlServerBytesWriter writes WKB compatible with geometry, not geography. So, we should explicitly set it for geography:
```suggestion
								var sqlGeometryWriter = new SqlServerBytesWriter
{
    IsGeography = true
};
```